src/rez/package_resources.py

```diff
@@ -310,7 +315,9 @@ class PackageFamilyResource(PackageRepositoryResource):
     A repository implementation's package family resource(s) must derive from
     this class. It must satisfy the schema `package_family_schema`.
     """
-    pass
+
+    def iter_packages(self) -> Iterator[Package]:
+        raise NotImplementedError
```

Comment:
All subclasses of `PackageFamilyResource` implement this method, so it appears to be considered an `abstractmethod` of `PackageFamilyResource`.   Defining it simplifies some type annotation problems.